### PR TITLE
Add Microdox Shield

### PR DESCRIPTION
**Pull request by careyk007**
I haven't had a chance to fully vet the keymap, but I based it off of the Corne's default with some changes due to missing pinkie columns. I'm waiting on switches to fully populate my board, but all switch positions are registering correctly.

This was pleasantly straight forward to get working, thanks for the great documentation!

### CODE REVIEW

File: app/boards/shields/microdox/microdox.dtsi [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+			, <&pro_micro_d 10 (GPIO_ACTIVE_HIGH | GPIO_PULL_DOWN)>
+			, <&pro_micro_d 8  (GPIO_ACTIVE_HIGH | GPIO_PULL_DOWN)>
+			, <&pro_micro_d 9  (GPIO_ACTIVE_HIGH | GPIO_PULL_DOWN)>
+			;
+		
+	};
+
+	bt_unpair_combo: bt_unpair_combo {
+		compatible = "zmk,bt-unpair-combo";
+	};
```
Comment (petejohanson) [inline app/boards/shields/microdox/microdox.dtsi:48]: We're deprecating this unpair combo stuff soon, we shouldn't add it for any new boards. Thanks!

File: app/boards/shields/microdox/microdox.keymap [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+ *
+ * SPDX-License-Identifier: MIT
+ */
+
+#include <behaviors.dtsi>
+#include <dt-bindings/zmk/keys.h>
+#include <dt-bindings/zmk/bt.h>
+
+/ {
+        keymap {
```
Comment (petejohanson) [inline app/boards/shields/microdox/microdox.keymap:12]: We're trying to be sure that all stock keymaps include at a minimum the `&bt BT_CLR` binding, and ideally the profile selection `&bt BT_SEL 0` bindings in some layer to make that usable out of the box. Can we find a spot here in one of the empty spots for this?

File: app/boards/shields/microdox/microdox.keymap [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+//             | GUI |     | SPC |   | ENT |     | ALT |
+                        bindings = <
+   &kp BANG  &kp ATSN  &kp HASH  &kp CURU  &kp PRCT    &kp CRRT  &kp AMPS &kp KMLT &kp LPRN &kp RPRN
+   &trans    &trans    &trans    &trans    &trans      &kp MINUS &kp EQL  &kp LBKT &kp RBKT &kp PIPE
+   &trans    &trans    &trans    &trans    &trans      &trans    &trans   &trans   &trans   &kp BSLH
+                       &kp LGUI  &trans    &kp SPC     &kp RET   &trans   &kp RALT
+                        >;
+                };
+
+                num_layer {
```
Comment (petejohanson) [inline app/boards/shields/microdox/microdox.keymap:56]: Until we figure out a sane way to do "tri layer state", this layer isn't reachable. If you want to leave it in here, I don't have a problem with it, but maybe a comment to that effect here would be good so users know.

File: app/boards/shields/microdox/microdox_left.overlay [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+		, <&pro_micro_a  2 GPIO_ACTIVE_HIGH>
+		, <&pro_micro_a  1 GPIO_ACTIVE_HIGH>
+		, <&pro_micro_a  0 GPIO_ACTIVE_HIGH>
+		, <&pro_micro_d 15 GPIO_ACTIVE_HIGH>
+		;
+};
+
+&bt_unpair_combo {
+	key-positions = <0 38>;
+};
```
Comment (petejohanson) [inline app/boards/shields/microdox/microdox_left.overlay:22]: Ditto, 'nix this.

File: app/boards/shields/microdox/microdox_right.overlay [hunk 5]
```diff
@@ -0,0 +1,26 @@
...
+		, <&pro_micro_a  0 GPIO_ACTIVE_HIGH>
+		, <&pro_micro_a  1 GPIO_ACTIVE_HIGH>
+		, <&pro_micro_a  2 GPIO_ACTIVE_HIGH>
+		, <&pro_micro_a  3 GPIO_ACTIVE_HIGH>
+		;
+};
+
+&bt_unpair_combo {
+	key-positions = <9 35>;
+};
```
Comment (petejohanson) [inline app/boards/shields/microdox/microdox_right.overlay:26]: Same

**Review by petejohanson — CHANGES_REQUESTED** [review]
Thanks for the new shield! A few miscellaneous comments and issues.

**Comment by careyk007** [comment]
@petejohanson Thanks for the review! I based all of these files on the corne shield (copied the folder and renamed everything microdox, then edited what needed tweaking). I'd be happy to make the changes you suggested here for the corne as well. I'll also take a stab at #181 with lessons learned.

File: app/boards/shields/microdox/microdox_left.overlay [hunk 6]
```diff
@@ -0,0 +1,18 @@
+/*
+ * Copyright (c) 2020 Pete Johanson
+ * Copyright (c) 2020 Kellen Carey
```
Comment (petejohanson) [inline app/boards/shields/microdox/microdox_left.overlay:3]: Missed a couple here:

```suggestion
 * Copyright (c) 2020 The ZMK Contributors
```

File: app/boards/shields/microdox/microdox_right.overlay [hunk 7]
```diff
@@ -0,0 +1,22 @@
+/*
+ * Copyright (c) 2020 Pete Johanson
+ * Copyright (c) 2020 Kellen Carey
```
Comment (petejohanson) [inline app/boards/shields/microdox/microdox_right.overlay:3]:
```suggestion
 * Copyright (c) 2020 The ZMK Contributors
```

**Review by petejohanson — CHANGES_REQUESTED** [review]

File: app/boards/shields/microdox/microdox_left.overlay (hunk 6 above)
Comment (careyk007) [reply app/boards/shields/microdox/microdox_left.overlay]: Shoot, thanks for catching that!

**Comment by careyk007** [comment]
Huh, I have no idea why changing the license would cause building the bluemicro iris_right shield to fail

**Comment by petejohanson** [comment]
@careyk007 The compiler *segfaulted* in that run... That definitely wasn't your doing. Re-running.

**Review by petejohanson — APPROVED** [review]
LGTM! Thanks.

**Pull request merged**